src/creature.cpp

```diff
@@ -673,10 +674,32 @@ void Creature::onDeath()
 				mostDamage = cb.total;
 				mostDamageCreature = attacker;
 			}
-			attacker->onAttackedCreatureKilled(this);
+
+			if (attacker != this) {
+				uint64_t gainExp = getGainedExperience(attacker);
+				if (Player* player = attacker->getPlayer()) {
+					Party* party = player->getParty();
+					if (party && party->isSharedExperienceActive() && party->isSharedExperienceEnabled()) {
+						if (Player* leader = party->getLeader()) {
```

Comment:
That's good enough, and I prefer it over declaring variables in if-statements, especially since party->getLeader() doesn't do anything heavy and most likely gets inlined by the compiler.